PixelHunter1995/SceneLib/Dog.cs

```diff
@@ -1,29 +1,30 @@
 ﻿using Microsoft.Xna.Framework;
-using Microsoft.Xna.Framework.Content;
 using Microsoft.Xna.Framework.Graphics;
+using PixelHunter1995.TilesetLib;
 
 namespace PixelHunter1995.SceneLib
 {
     class Dog : IDrawable
     {
-        public int X { get; private set; }
-        public int Y { get; private set; }
-        public int Width { get; private set; }
-        public int Height { get; private set; }
-        private int gid;
+        private readonly int X;
+        private readonly int Y;
+        private readonly int Width;
+        private readonly int Height;
+        private readonly int Gid;
+        private readonly Tileset Tileset;
 
-        public Dog(int x, int y, int width, int height, int gid)
+        public Dog(int x, int y, int width, int height, int gid, Tileset tileset)
         {
             this.X = x;
             this.Y = y;
             this.Width = width;
             this.Height = height;
-            this.gid = gid;
+            this.Gid = gid;
+            this.Tileset = tileset;
         }
-        public void Draw(GraphicsDeviceManager graphics, SpriteBatch spriteBatch, Tileset tileset)
+        public void Draw(GraphicsDeviceManager graphics, SpriteBatch spriteBatch)
         {
-            Rectangle destinationRectangle = new Rectangle(X, Y, Width, Height);
-            tileset.DrawTile(spriteBatch, destinationRectangle, gid);
+            Tileset.Draw(spriteBatch, new Vector2(X, Y), Gid);
```

Comment:
Why no width/height? Do we decide there will be no scaling?

Reply:
Created new scaling parameter.